app/utils.py

```diff
@@ -1,6 +1,45 @@
 import re
 
 
+def list_pointers(input_data, pointer=None):
+    """
+    Recursive function which lists all available pointers in a json structure
+    :param input_data: the input data to search
+    :param pointer: the current pointer
+    :return: generator of the json pointer paths
+    """
+    if isinstance(input_data, dict) and input_data:
+        for k, v in input_data.items():
+            yield pointer + '/' + k if pointer else '/' + k
+            yield from list_pointers(v, pointer + '/' + k if pointer else '/' + k)
+    elif isinstance(input_data, list) and input_data:
+        for index, item in enumerate(input_data):
+            yield from list_pointers(item, '{}/{}'.format(pointer, index))
```

Comment:
Does there need to be an else clause here to handle lists of strings / ints etc?

Reply:
Not an else, but another yield - I've extended the tests to cover this.